### PR TITLE
avm2: Fix `in` operator not handling attributes in XML object

### PR DESCRIPTION
**Pull request by sleepycatcoding**
This makes expressions like
```
var x = <root val="hello"/>
"@val" in x; // Was false before this PR.
```
work.

### CODE REVIEW

File: core/src/avm2/object/xml_object.rs [hunk 1]
```diff
@@ -365,6 +365,15 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
         self.0.read().base.has_own_dynamic_property(name)
     }
 
+    fn has_property_via_in(
+        self,
+        activation: &mut Activation<'_, 'gc>,
+        name: &Multiname<'gc>,
+    ) -> Result<bool, Error<'gc>> {
+        let multiname = handle_input_multiname(name.clone(), activation);
+        Ok(self.has_own_property(&multiname))
```
Comment (evilpie) [inline core/src/avm2/object/xml_object.rs:374]: This doesn't seem correct. `in` needs to check whole chain. Consider `"propertyIsEnumerable" in xml` vs `xml.hasOwnProperty("propertyIsEnumerable")`.

**Review by evilpie — CHANGES_REQUESTED** [review]

File: core/src/avm2/object/xml_object.rs (hunk 1 above)
Comment (sleepycatcoding) [reply core/src/avm2/object/xml_object.rs:374]: Would changing to `self.has_property` be enough?

Comment (sleepycatcoding) [reply core/src/avm2/object/xml_object.rs]: > This doesn't seem correct. `in` needs to check whole chain. Consider `"propertyIsEnumerable" in xml` vs `xml.hasOwnProperty("propertyIsEnumerable")`.

At the moment `"propertyIsEnumerable" in xml` seems to return `false` on master too.

Comment (evilpie) [reply core/src/avm2/object/xml_object.rs]: > Would changing to `self.has_property` be enough, like how `has_property_via_in` does by default?

I think that might work? I think it should end up calling the XML's `has_own_property`.

> At the moment `"propertyIsEnumerable" in xml` seems to return `false` on master too.

It's `true` in Flash.

File: tests/tests/swfs/avm2/xml_has_property_via_in/Test.as [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+trace("@a" in a);
+trace("");
+
+trace("0 in a");
+trace(0 in a);
+trace("");
+
+trace("1 in a");
+trace(1 in a);
+trace("");
```
Comment (evilpie) [inline tests/tests/swfs/avm2/xml_has_property_via_in/Test.as:38]: Can you add a test for isPropertyEnumerable?

Comment (sleepycatcoding) [reply tests/tests/swfs/avm2/xml_has_property_via_in/Test.as]: This currently fails, but seems to work with your PR #15120.

Comment (evilpie) [reply tests/tests/swfs/avm2/xml_has_property_via_in/Test.as]: We can land #1520 first and use this as a test for it.

**Pull request merged**